src/Orleans/Utils/Factory.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace Orleans
+{
+    /// <summary>
+    /// Creates an instance of <typeparamref name="TInstance"/>.
+    /// </summary>
+    /// <typeparam name="TInstance"></typeparam>
+    /// <returns></returns>
+    public delegate TInstance Factory<out TInstance>();
```

Comment:
After @jdom pointed to me to FactoryUtilities that is exactly what I was looking for (and missed somehow). Renaming to ```ObjectFactory``` can help if we submit this as a PR to the DI Abstraction repo, but it is not a big deal if we leave it as ```Factory```.